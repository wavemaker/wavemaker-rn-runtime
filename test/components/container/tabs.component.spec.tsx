import React, { createRef, ReactNode, useRef } from 'react';
import { ScrollView, Text, TouchableOpacity } from 'react-native';
import {
  render,
  fireEvent,
  cleanup,
  screen,
} from '@testing-library/react-native';
import WmTabs from '@wavemaker/app-rn-runtime/components/container/tabs/tabs.component';
import WmTabpane from '@wavemaker/app-rn-runtime/components/container/tabs/tabpane/tabpane.component';
import WmTabheader from '@wavemaker/app-rn-runtime/components/container/tabs/tabheader/tabheader.component';

const getBasicTabs = () => {
  const tabs = createRef<WmTabs>();
  const tab1 = createRef<WmTabpane>();
  const tab2 = createRef<WmTabpane>();
  const tab3 = createRef<WmTabpane>();
  return {
    tree: render(
      <WmTabs name="test_tabs" ref={tabs}>
        <WmTabpane name="tab1" title="Red" ref={tab1}></WmTabpane>
        <WmTabpane name="tab2" title="Green" ref={tab2}></WmTabpane>
        <WmTabpane name="tab3" title="Blue" ref={tab3}></WmTabpane>
      </WmTabs>
    ),
    ref: {
      tabs,
      tab1,
      tab2,
      tab3,
    },
  };
};

const timer = (time = 100) =>
  new Promise((resolve: any, reject) => {
    setTimeout(() => resolve(), time);
  });

describe('Test Tabs component', () => {
  test('should render the Tabs component', () => {
    const {
      tree,
      ref: { tabs },
    } = getBasicTabs();
    expect(tree).not.toBeNull();
    expect(tree).toBeDefined();
    expect(tree).toMatchSnapshot();
    expect(tabs.current).not.toBeNull();
  });

  test('should select the first tab pane by default', () => {
    const {
      ref: { tabs, tab1 },
    } = getBasicTabs();
    expect(tab1.current).not.toBeNull();
    expect(tabs.current?.selectedTabPane === tab1.current?.proxy).toBeTruthy();
  });

  test('should select the tab pane based on header pressed', async () => {
    const {
      tree,
      ref: { tabs, tab1, tab3 },
    } = getBasicTabs();
    expect(tab1.current).not.toBeNull();
    expect(tabs.current?.selectedTabPane === tab1.current?.proxy).toBeTruthy();
    const header3 = tree.getByText('Blue');
    fireEvent(header3, 'press');
    await timer(1000);
    expect(tab3.current).not.toBeNull();
    expect(tabs.current?.selectedTabPane === tab3.current?.proxy).toBeTruthy();
  });

  test('should check the goTo method', async () => {
    const {
      tree,
      ref: { tabs, tab1, tab3 },
    } = getBasicTabs();
    expect(tab1.current).not.toBeNull();
    expect(tabs.current?.selectedTabPane === tab1.current?.proxy).toBeTruthy();
    tabs.current?.goToTab(2);
    await timer(1000);
    expect(tab3.current).not.toBeNull();
    expect(tabs.current?.selectedTabPane === tab3.current?.proxy).toBeTruthy();
    tabs.current?.goToTab(-1);
    expect(tabs.current?.selectedTabPane === tab3.current?.proxy).toBeTruthy();
    tabs.current?.goToTab(4);
    expect(tabs.current?.selectedTabPane === tab3.current?.proxy).toBeTruthy();
    tabs.current?.goToTab(0);
    await timer(1000);
    expect(tabs.current?.selectedTabPane === tab1.current?.proxy).toBeTruthy();
  });

  test('should select the tab pane with select method', async () => {
    const {
      tree,
      ref: { tabs, tab1, tab3 },
    } = getBasicTabs();
    expect(tab1.current).not.toBeNull();
    expect(tabs.current?.selectedTabPane === tab1.current?.proxy).toBeTruthy();
    expect(tab3.current).not.toBeNull();
    (tab3.current?.proxy as any).select();
    await timer(1000);
    expect(tabs.current?.selectedTabPane === tab3.current?.proxy).toBeTruthy();
  });

  test('should render the tabs when content is from partial', async () => {
    const tab2 = createRef<WmTabpane>();
    const onLoadMock = jest.fn();

    const tree = render(
      <WmTabs name="test_tabs">
        <WmTabpane name="tab1" title="Red"></WmTabpane>
        <WmTabpane
          name="tab2"
          title="Green"
          ref={tab2}
          onLoad={onLoadMock}
          renderPartial={(props, onLoad) => {
            onLoad();
            return <Text>TEST_COMPONENT</Text>;
          }}
        ></WmTabpane>
      </WmTabs>
    );
    timer(1000);
    expect(tree.getByText('TEST_COMPONENT')).toBeTruthy();
    expect(onLoadMock).toHaveBeenCalled();
  });

  test('should render skeleton loader when showskeleton is "true"', () => {
    const tree = render(
      <WmTabs name="test_tabs" showskeleton={true}>
        <WmTabpane name="tab1" title="Red"></WmTabpane>
        <WmTabpane
          name="tab2"
          title="Green"
          renderPartial={() => {
            return <Text>TEST_COMPONENT</Text>;
          }}
        ></WmTabpane>
      </WmTabs>
    );
    expect(tree).toMatchSnapshot();
    expect(screen.queryByText('Red')).toBeNull();
    expect(screen.queryByText('Green')).toBeNull();
  });

  test('handles show property correctly', async () => {
    const ref = createRef();
    const tree = render(
      <WmTabs name="test_Popover" show={true} ref={ref}>
        <WmTabpane name="tab1" title="Red"></WmTabpane>
        <WmTabpane
          name="tab2"
          title="Green"
          renderPartial={() => {
            return <Text>TEST_COMPONENT</Text>;
          }}
        ></WmTabpane>
      </WmTabs>
    );

    expect(tree.root.props.style.width).not.toBe(0);
    expect(tree.root.props.style.height).not.toBe(0);

    ref.current.proxy.show = false;

    await timer(300);

    expect(tree.root.props.style.width).toBe(0);
    expect(tree.root.props.style.height).toBe(0);
  });

  it('should handle tablayout change event', () => {
    const ref = createRef();
    const tree = render(
      <WmTabs name="test_Popover" ref={ref}>
        <WmTabpane name="tab1" title="Red"></WmTabpane>
        <WmTabpane
          name="tab2"
          title="Green"
          renderPartial={() => {
            return <Text>TEST_COMPONENT</Text>;
          }}
        ></WmTabpane>
      </WmTabs>
    );
    const nativeEvent = {
      layout: {
        width: 100,
        height: 100,
      },
    };
    const viewEle = tree.root.children[1];
    fireEvent(viewEle, 'layout', {
      nativeEvent: nativeEvent,
    });

    expect(ref.current.proxy.tabLayout.toString()).toBe(nativeEvent.toString());
  });

  it('should set tab pane heights when layout change event is triggered', () => {
    const ref = createRef();
    const tree = render(
      <WmTabs name="test_Popover" ref={ref}>
        <WmTabpane name="tab1" title="Red"></WmTabpane>
        <WmTabpane
          name="tab2"
          title="Green"
          renderPartial={() => {
            return <Text>TEST_COMPONENT</Text>;
          }}
        ></WmTabpane>
      </WmTabs>
    );
    const nativeEvent = {
      layout: {
        width: 100,
        height: 100,
      },
    };
    const viewEle = tree.UNSAFE_getAllByType(WmTabpane)[0].parent;
    fireEvent(viewEle, 'layout', {
      nativeEvent: nativeEvent,
    });

    expect(ref.current.proxy.tabPaneHeights.toString()).toBe(
      nativeEvent.layout.height.toString()
    );
  });

  it('should handle tabheader layout change events', () => {
    const ref = createRef();
    const tree = render(<WmTabheader name="test_Popover" ref={ref} data={[
      {title: 'tab1', icon: 'fa fa-edit', key: 'tab1'},
      {title: 'tab2', icon: 'fa fa-edit', key: 'tab2'},
      {title: 'tab3', icon: 'fa fa-edit', key: 'tab3'},
    ]}/>);
    const nativeEvent = {
      layout: {
        width: 100,
        height: 100,
      },
    };

    ref.current.proxy.setHeaderPanelPositon({ nativeEvent: nativeEvent });
    ref.current.proxy.setHeaderPositon(1, { nativeEvent: nativeEvent });

    expect(ref.current.proxy.headerPanelLayout).toMatchObject(
      nativeEvent.layout
    );

    expect(ref.current.proxy.headersLayout[1]).toMatchObject(
      nativeEvent.layout
    );
  });

  it('should disable scroll when shouldScroll prop is false', () => {
    const ref = createRef();
    const tree = render(<WmTabheader name="test_Popover" shouldScroll={false} ref={ref} data={[
      {title: 'tab1', icon: 'fa fa-edit', key: 'tab1'},
      {title: 'tab2', icon: 'fa fa-edit', key: 'tab2'},
      {title: 'tab3', icon: 'fa fa-edit', key: 'tab3'},
    ]}/>);
    const nativeEvent = {
      layout: {
        width: 100,
        height: 100,
      },
    };

    ref.current.proxy.setHeaderPanelPositon({ nativeEvent: nativeEvent });
    ref.current.proxy.setHeaderPositon(1, { nativeEvent: nativeEvent });

    const scrollComponent = tree.UNSAFE_getByType(ScrollView);

    expect(ref.current.proxy.headerPanelLayout).toMatchObject(
      nativeEvent.layout
    );

    expect(ref.current.proxy.headersLayout[1]).toMatchObject(
      nativeEvent.layout
    );

    expect(scrollComponent.props.scrollEnabled).toBe(false)
  });

  // test('should navigate to next and previous', async () => {
  //   const {tree, ref: {tabs, tab1, tab2, tab3}} = getBasicTabs();
  //   expect(tabs.current?.selectedTabPane === tab1.current?.proxy).toBeTruthy();
  //   tabs.current?.next();
  //   await timer(1000);
  //   expect(tabs.current?.selectedTabPane === tab2.current?.proxy).toBeTruthy();
  //   tabs.current?.next();
  //   await timer(1000);
  //   expect(tabs.current?.selectedTabPane === tab3.current?.proxy).toBeTruthy();
  //   tabs.current?.prev();
  //   await timer(1000);
  //   expect(tabs.current?.selectedTabPane === tab2.current?.proxy).toBeTruthy();
  // });
});
