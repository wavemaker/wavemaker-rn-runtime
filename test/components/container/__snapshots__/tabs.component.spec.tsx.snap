// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tabs component should render skeleton loader when showskeleton is "true" 1`] = `
<View
  style={
    [
      {
        "borderBottomWidth": 1,
        "borderColor": "#cccccc",
        "borderStyle": "solid",
        "elevation": 0,
        "minHeight": 240,
      },
      {
        "borderBottomWidth": 0,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "width": "100%",
      }
    }
  />
  <View
    collapsable={false}
    onLayout={[Function]}
    style={
      {
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "minWidth": "100%",
          "overflow": "visible",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flexDirection": "row",
                "flexGrow": 1,
                "justifyContent": "center",
                "minHeight": 48,
                "minWidth": 80,
                "paddingBottom": 16,
                "paddingTop": 16,
                "paddingVertical": 12,
              },
              {
                "backgroundColor": "#ffffff",
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              {
                "height": 0,
                "width": 0,
              }
            }
          />
          <View
            onLayout={[Function]}
            style={
              {
                "backgroundColor": "#eeeeee",
                "borderRadius": 4,
                "height": 16,
                "overflow": "hidden",
                "width": "80%",
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        <View
          onLayout={[Function]}
          style={
            [
              {
                "backgroundColor": "#ffffff",
                "flexDirection": "row",
                "flexGrow": 1,
                "justifyContent": "center",
                "minHeight": 48,
                "minWidth": 80,
                "paddingBottom": 16,
                "paddingTop": 16,
                "paddingVertical": 12,
              },
              null,
            ]
          }
        >
          <View
            onLayout={[Function]}
            style={
              {
                "height": 0,
                "width": 0,
              }
            }
          />
          <View
            onLayout={[Function]}
            style={
              {
                "backgroundColor": "#eeeeee",
                "borderRadius": 4,
                "height": 16,
                "overflow": "hidden",
                "width": "80%",
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#4263eb",
          "height": 2,
          "marginTop": -2,
          "transform": [
            {
              "translateX": 0,
            },
            {
              "scaleX": 0,
            },
          ],
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      {
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "flexWrap": "nowrap",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "width": "100%",
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            {
              "height": 0,
              "width": 0,
            }
          }
        />
        <RCTScrollView
          collapsable={false}
          handlerTag={24}
          handlerType="NativeViewGestureHandler"
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onScroll={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
            }
          }
          waitFor={
            [
              {
                "current": null,
              },
            ]
          }
        >
          <View />
        </RCTScrollView>
      </View>
      <View
        onLayout={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "width": "100%",
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            {
              "height": 0,
              "width": 0,
            }
          }
        />
        <RCTScrollView
          collapsable={false}
          handlerTag={25}
          handlerType="NativeViewGestureHandler"
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onScroll={[Function]}
          style={
            {
              "backgroundColor": "#ffffff",
            }
          }
          waitFor={
            [
              {
                "current": null,
              },
            ]
          }
        >
          <View>
            <Text>
              TEST_COMPONENT
            </Text>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test Tabs component should render the Tabs component 1`] = `
[
  <View
    onLayout={[Function]}
    style={
      {
        "height": 0,
        "width": 0,
      }
    }
  />,
  <View
    style={
      {
        "borderBottomWidth": 1,
        "borderColor": "#cccccc",
        "borderStyle": "solid",
        "elevation": 0,
        "minHeight": 240,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        {
          "width": "100%",
        }
      }
    />
    <View
      onLayout={[Function]}
      style={
        {
          "height": 0,
          "width": 0,
        }
      }
    />
    <View
      accessible={false}
      style={
        {
          "overflow": "hidden",
          "zIndex": 16,
        }
      }
      testID="test_tabs_headers_tabheader"
    >
      <RCTScrollView
        horizontal={true}
        onLayout={[Function]}
        scrollEnabled={false}
        showsHorizontalScrollIndicator={false}
      >
        <View>
          <View>
            <View
              style={
                {
                  "backgroundColor": "#ffffff",
                  "flexDirection": "row",
                  "minWidth": "100%",
                  "overflow": "visible",
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "flexGrow": 1,
                    },
                  ]
                }
                testID="test_tabs_headers_0"
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    style={
                      [
                        {
                          "backgroundColor": "#ffffff",
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "justifyContent": "center",
                          "minHeight": 48,
                          "minWidth": 80,
                          "paddingBottom": 16,
                          "paddingTop": 16,
                          "paddingVertical": 12,
                        },
                        {
                          "flexGrow": undefined,
                        },
                        {
                          "backgroundColor": "#ffffff",
                        },
                      ]
                    }
                  >
                    <View
                      onLayout={[Function]}
                      style={
                        {
                          "height": 0,
                          "width": 0,
                        }
                      }
                    />
                    <View>
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "alignSelf": "flex-start",
                            "flexDirection": "row",
                          }
                        }
                        testID="non_animatableView"
                      />
                    </View>
                    <Text
                      accessible={false}
                      numberOfLines={1}
                      style={
                        [
                          {
                            "color": "#aaaaaa",
                            "fontFamily": "Roboto",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "overflow": "visible",
                          },
                          {
                            "color": "#4263eb",
                          },
                        ]
                      }
                      testID="test_tabs_headers_0_title"
                    >
                      Red
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "flexGrow": 1,
                    },
                  ]
                }
                testID="test_tabs_headers_1"
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    style={
                      [
                        {
                          "backgroundColor": "#ffffff",
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "justifyContent": "center",
                          "minHeight": 48,
                          "minWidth": 80,
                          "paddingBottom": 16,
                          "paddingTop": 16,
                          "paddingVertical": 12,
                        },
                        {
                          "flexGrow": undefined,
                        },
                        null,
                      ]
                    }
                  >
                    <View
                      onLayout={[Function]}
                      style={
                        {
                          "height": 0,
                          "width": 0,
                        }
                      }
                    />
                    <View>
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "alignSelf": "flex-start",
                            "flexDirection": "row",
                          }
                        }
                        testID="non_animatableView"
                      />
                    </View>
                    <Text
                      accessible={false}
                      numberOfLines={1}
                      style={
                        [
                          {
                            "color": "#aaaaaa",
                            "fontFamily": "Roboto",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "overflow": "visible",
                          },
                          null,
                        ]
                      }
                      testID="test_tabs_headers_1_title"
                    >
                      Green
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": false,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "flexGrow": 1,
                    },
                  ]
                }
                testID="test_tabs_headers_2"
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    style={
                      [
                        {
                          "backgroundColor": "#ffffff",
                          "flexDirection": "row",
                          "flexGrow": 1,
                          "justifyContent": "center",
                          "minHeight": 48,
                          "minWidth": 80,
                          "paddingBottom": 16,
                          "paddingTop": 16,
                          "paddingVertical": 12,
                        },
                        {
                          "flexGrow": undefined,
                        },
                        null,
                      ]
                    }
                  >
                    <View
                      onLayout={[Function]}
                      style={
                        {
                          "height": 0,
                          "width": 0,
                        }
                      }
                    />
                    <View>
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "alignSelf": "flex-start",
                            "flexDirection": "row",
                          }
                        }
                        testID="non_animatableView"
                      />
                    </View>
                    <Text
                      accessible={false}
                      numberOfLines={1}
                      style={
                        [
                          {
                            "color": "#aaaaaa",
                            "fontFamily": "Roboto",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "overflow": "visible",
                          },
                          null,
                        ]
                      }
                      testID="test_tabs_headers_2_title"
                    >
                      Blue
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#4263eb",
                  "height": 2,
                  "marginTop": -2,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                    {
                      "scaleX": 0,
                    },
                  ],
                  "width": 100,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundImage": "",
                    "backgroundPosition": "center",
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "100% 100%",
                    "display": "none",
                    "transform": [
                      {
                        "scaleX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    {
                      "display": "none",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
            "width": "100%",
          },
          {
            "maxHeight": undefined,
            "overflow": "hidden",
          },
          {},
        ]
      }
    >
      <View
        collapsable={false}
        onLayout={[Function]}
        style={
          {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "transform": [
              {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            [
              {
                "width": "100%",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignSelf": "flex-start",
                "height": 1000000,
                "width": "100%",
              }
            }
          >
            <View
              onLayout={[Function]}
              style={
                {
                  "alignSelf": "flex-start",
                  "width": "100%",
                }
              }
            >
              <View
                onLayout={[Function]}
                style={
                  {
                    "height": 0,
                    "width": 0,
                  }
                }
              />
              <RCTScrollView
                collapsable={false}
                handlerTag={2}
                handlerType="NativeViewGestureHandler"
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onScroll={[Function]}
                style={
                  {
                    "backgroundColor": "#ffffff",
                  }
                }
                waitFor={
                  [
                    {
                      "current": null,
                    },
                  ]
                }
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            [
              {
                "width": "100%",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignSelf": "flex-start",
                "height": 1000000,
                "width": "100%",
              }
            }
          >
            <View
              onLayout={[Function]}
              style={
                {
                  "alignSelf": "flex-start",
                  "width": "100%",
                }
              }
            >
              <View
                onLayout={[Function]}
                style={
                  {
                    "height": 0,
                    "width": 0,
                  }
                }
              />
              <RCTScrollView
                collapsable={false}
                handlerTag={3}
                handlerType="NativeViewGestureHandler"
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onScroll={[Function]}
                style={
                  {
                    "backgroundColor": "#ffffff",
                  }
                }
                waitFor={
                  [
                    {
                      "current": null,
                    },
                  ]
                }
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            [
              {
                "width": "100%",
              },
              null,
            ]
          }
        >
          <View
            style={
              {
                "alignSelf": "flex-start",
                "height": 1000000,
                "width": "100%",
              }
            }
          >
            <View
              onLayout={[Function]}
              style={
                {
                  "alignSelf": "flex-start",
                  "width": "100%",
                }
              }
            >
              <View
                onLayout={[Function]}
                style={
                  {
                    "height": 0,
                    "width": 0,
                  }
                }
              />
              <RCTScrollView
                collapsable={false}
                handlerTag={4}
                handlerType="NativeViewGestureHandler"
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onScroll={[Function]}
                style={
                  {
                    "backgroundColor": "#ffffff",
                  }
                }
                waitFor={
                  [
                    {
                      "current": null,
                    },
                  ]
                }
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
